src/Symfony/Component/VarExporter/Internal/Registry.php

```diff
@@ -40,7 +40,11 @@ public static function unserialize($objects, $serializables)
 
         try {
             foreach ($serializables as $k => $v) {
-                $objects[$k] = unserialize($v);
+                try {
+                    $objects[$k] = unserialize($v);
+                } catch (\UnserializationFailedException $e) {
+                    throw $e->getPrevious() instanceof ClassNotFoundException ? $e->getPrevious() : $e;
```

Comment:
Same here: in this method, we register an "unserialize_callback_func" to gracefully detect class-not-found. The RFC breaks the logic and we need those extra lines to restore the expected behavior.